src/ControlSystem/Measurements/BNSCenterOfMass.hpp

```diff
@@ -0,0 +1,182 @@
+                                         inv_det_jacobian, tilde_d, x_grid);
+
+      // Reduction
+      auto& my_proxy =
+          Parallel::get_parallel_component<ParallelComponent>(cache);
+      // We need a place to run RunCallback on... this does not need to be
+      // the control system using the CoM data.
+      auto& reduction_target_proxy = Parallel::get_parallel_component<
+          ControlComponent<Metavariables, tmpl::front<ControlSystems>>>(cache);
+      Parallel::ReductionData<
+          Parallel::ReductionDatum<LinkedMessageId<double>,
+                                   funcl::AssertEqual<>>,
+          Parallel::ReductionDatum<double, funcl::Plus<>>,
+          Parallel::ReductionDatum<double, funcl::Plus<>>,
+          Parallel::ReductionDatum<std::array<double, 3>, funcl::Plus<>>,
+          Parallel::ReductionDatum<std::array<double, 3>, funcl::Plus<>>>
+          reduction_data{measurement_id, mass_a, mass_b, first_moment_a,
+                         first_moment_b};
+      Parallel::contribute_to_reduction<
+          PostReductionSendBNSStarCentersToControlSystem<ControlSystems>>(
+          std::move(reduction_data), my_proxy, reduction_target_proxy);
+    }
+  };
+  /// List of submeasurements used by this measurement -- only FindTwoCenters
+  /// here.
+  using submeasurements = tmpl::list<FindTwoCenters>;
+};
+
+/// Action called after reduction of the center of mass data.
+/// MassA, MassB, FirstMomentA, FirstMomentB will contain the reduced data
+/// for the integral of the density (and its first moment) in the x>0 (A label)
+/// and x<0 (B label) regions.
+/// This action calculates the center of mass in each region, and sends the
+/// result to the control system.
+template <typename ControlSystems>
+struct PostReductionSendBNSStarCentersToControlSystem {
+  template <typename ParallelComponent, typename DbTags, typename Metavariables,
+            typename ArrayIndex>
+  static void apply(db::DataBox<DbTags>& /*box*/,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const LinkedMessageId<double>& measurement_id,
+                    const double& mass_a, const double& mass_b,
+                    const std::array<double, 3>& first_moment_a,
+                    const std::array<double, 3>& first_moment_b) {
+    // Function called after reduction of the CoM data.
+    // 1. Calculate CoM from integrals
+    std::array<double, 3> center_a = first_moment_a;
+    std::array<double, 3> center_b = first_moment_b;
+    for (size_t d = 0; d < 3; d++) {
+      center_a[d] = center_a[d] / mass_a;
+      center_b[d] = center_b[d] / mass_b;
+    }
+    const auto center_databox = db::create<
+        db::AddSimpleTags<Tags::NeutronStarCenter<::domain::ObjectLabel::A>,
+                          Tags::NeutronStarCenter<::domain::ObjectLabel::B>>>(
+        center_a, center_b);
```

Comment:
If you `#include Utilities/StdArrayHelpers.hpp` we have an overload of the `opterator/()` that works with std::arrays. Then you can just put here `first_moment_a / mass_a, first_moment_b / mass_b`

Reply:
Done